File: 2639-separate-the-digits-in-an-array/separate-the-digits-in-an-array.cpp
class Solution {
public:
    vector<int> separateDigits(vector<int>& nums) {
        vector<int> res;
        for (int i = nums.size() - 1; i >= 0; i--) {
            int n = nums[i];
            while (n) {
                res.push_back(n % 10);
                n /= 10;
            }
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

File: 2639-separate-the-digits-in-an-array/separate-the-digits-in-an-array.js

var separateDigits = function (nums) {
    const resArr = []
    for (let i = nums.length - 1; i >= 0; i--) {
        let n = nums[i]
        while (n) {
            resArr.push(n % 10)
            n = Math.floor(n / 10)
        }
    }
    return resArr.reverse()
};